### ISSUE TITLE
Grouping by Option[List[String]] fails with must appear in the GROUP BY clause or be used in an aggregate function

### ISSUE THREAD

**Issue by disblader** [tminglei/slick-pg#449]
Hello,

I'm having an issue trying to group by a nullable array column with a default. 

I have a table like this

```scala
trait TestTableSpec { this: TestSchema with PgArraySupport =>

  import api._

  class TestTable(tag: Tag) extends Table[Test](tag, "test") {

    def id = column[Int]("id")
    def timestamp = column[LocalDateTime]("timerino")
    def one = column[Int]("one")
    def optionInt = column[Option[Int]]("two")
    def three = column[List[Int]]("three")
    def four = column[Option[List[String]]]("four")

    def * = (id, timestamp, one, optionInt, three, four) <> ((Test.apply _).tupled, Test.unapply)

  }

  val tests = lifted.TableQuery[TestTable]

}

case class Test(id: Int, time: LocalDateTime, one: Int, optionInt: Option[Int], three: List[Int], four: Option[List[String]])
```

And I'm trying to perform a query as such:

```scala
  def caseTest = {
    db.run(
        tests.groupBy(table =>
          table.four.getOrElse(List("No"))

//          The below works
//          table.four

// Tried the below as well, but results in "Caught exception while computing default value for Rep[Option[_]].getOrElse -- This cannot be done lazily when the value is needed on the database side"
//          (Case If table.four.isDefined Then table.four.get Else List("No")).unnest
        ).map(q => q._1 -> q._2.size).result

    )
  }
```

However, the query results in the following:

```
[info] 1418 [scala-execution-context-global-15] DEBUG slick.basic.BasicBackend.action - #1: result [select coalesce("four", ?), count(1) from "test" group by coalesce("four", ?)]
[info] 1419 [test-2] DEBUG slick.jdbc.JdbcBackend.statement - Preparing statement: select coalesce("four", ?), count(1) from "test" group by coalesce("four", ?)
[info] 1425 [test-2] DEBUG slick.jdbc.JdbcBackend.statement - Executing prepared statement: HikariProxyPreparedStatement@809193186 wrapping select coalesce("four", '{"No"}'), count(1) from "test" group by coalesce("four", '{"No"}')
[info] 1425 [test-2] DEBUG s.j.J.statementAndParameter - Executing prepared statement: HikariProxyPreparedStatement@809193186 wrapping select coalesce("four", '{"No"}'), count(1) from "test" group by coalesce("four", '{"No"}')
[info] 1430 [test-2] DEBUG slick.jdbc.JdbcBackend.parameter - /--------+--------\
[info] 1430 [test-2] DEBUG slick.jdbc.JdbcBackend.parameter - | 1      | 2      |
[info] 1430 [test-2] DEBUG slick.jdbc.JdbcBackend.parameter - | Array  | Array  |
[info] 1430 [test-2] DEBUG slick.jdbc.JdbcBackend.parameter - |--------+--------|
[info] 1430 [test-2] DEBUG slick.jdbc.JdbcBackend.parameter - | {"No"} | {"No"} |
[info] 1430 [test-2] DEBUG slick.jdbc.JdbcBackend.parameter - \--------+--------/
[error] org.postgresql.util.PSQLException: ERROR: column "test.four" must appear in the GROUP BY clause or be used in an aggregate function
```

Copying the exact query from the logs and running it in postgres console works just fine, so I'm a little stumped.

Thanks for your time.

**Comment by tminglei** [comment]
Strange, the generated sql looks fine. I'll dig into it.

**Comment by tminglei** [comment]
Seems related to pgjdbc/pgjdbc#906.

**Comment by disblader** [comment]
I tried using `slick.lifter.LiteralColumn` syntax to provide the value as well, but it did not change anything unfortunately. I noticed that in `PgArrayJdbcTypes` it has the following:
```scala
 override def hasLiteralForm: Boolean = false
```
Can arrays not be literals via slick-pg?

**Comment by tminglei** [comment]
Well, my fault. I'll adjust the codes.

**Comment by tminglei** [comment]
@disblader fixed, pls help check. _Thanks! 👍_

**Comment by disblader** [comment]
Awesome, thanks for the prompt fix. I'll build latest version and try it out. 

When do you think this will make it to maven?

**Comment by tminglei** [comment]
Won't be too long.

**Comment by tminglei** [comment]
Released in v0.17.3
